UT-Applicataion-Development-Association/pharmaceutical
# Add basic orm

## Related Issue



## Type of Change


## Major Changes
- Add database  under `Backend/server/db/varianet_database.db`

- Add ORM models under `Backend/server/db/model.py`

- Add database setup under `Backend/server/__init__.py`

- Add test route related with database under `Backend/server/routes/signup`

## Steps for Testing
Starting the server and database, submit a post request to localhost:8080/signup.  This should insert a row to the Symptom


## Additional Context
Improvements need to be done (Priority from High to Low)
- Make sure all Model instance can be successfully created

- Make sure foreign constraints are working

- Adding relationship to ORM models [Officail Documentation](https://flask-sqlalchemy.palletsprojects.com/en/2.x/models/)

- Make sure the git-secret is working, so that we can encrypt the credentials without push plaintext directly to the repository.  Some Useful documentation: 
    - [Offical Documentation](https://git-secret.io/)
   
    - [Chinese Blog](https://tautcony.xyz/2020/03/07/git-secret/)

    - [Chinese Blog2](https://blog.csdn.net/JENREY/article/details/105162201)

- If possible, we should also improve the import structure in the service.  The signup service now needs to import the app from the server, from my point of view, this is quite weird

- In the end, we should also add more files on the .gitignore, including idea folder, DS_STORE, and **varient_database.db**
